table/data_block_index_format.h

```diff
@@ -0,0 +1,26 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
```

Comment:
You essentially made the last 4 bytes of a data block to be a "data block footer". I am wondering if we should call this file data_block_footer_format instead? I don't have a strong opinion on this either way.